src/Orchard.Web/Modules/Orchard.OpenId/OwinMiddlewares/Google.cs

```diff
@@ -27,7 +29,7 @@ public class Google : IOwinMiddlewareProvider {
             var authenticationOptions = new GoogleOAuth2AuthenticationOptions {
                 ClientId = settings.ClientId,
                 ClientSecret = settings.ClientSecret,
-                CallbackPath = new PathString(settings.CallbackPath)
+                CallbackPath = new PathString(GetCallbackPath(workContext, settings))
```

Comment:
Isn't it fine if the users adds the prefix by themself? We can just document that the url they enter needs to contain the tenant prefix if there is one. I would prefer this option if it works.